onnx/backend/test/case/node/rotaryembedding.py

```diff
@@ -0,0 +1,162 @@
+# Copyright (c) ONNX Project Contributors
+#
+# SPDX-License-Identifier: Apache-2.0
+from __future__ import annotations
+
+import numpy as np
+
+import onnx
+from onnx.backend.test.case.base import Base
+from onnx.backend.test.case.node import expect
+
+
+def compute_rotary_embedding(
+    input: np.ndarray,
+    position_ids: np.ndarray,
+    sin_cache: np.ndarray,
+    cos_cache: np.ndarray,
+    interleaved: int = 0,
+    rotary_embedding_dim: int = 0,
+    num_heads: int = 0,
+) -> np.ndarray:
+
+    # First ensure input has shape [batch_size, num_heads, seq_len, head_size]
+    batch_size = input.shape[0]
+    sequence_length = input.shape[1]
+    if len(input.shape) == 3:
+        hidden_size = input.shape[2]
+        assert num_heads != 0
+        head_size = int(hidden_size / num_heads)
```

Comment:
If hidden_size is required to be a multiple of num_heads, maybe we can assert/check that too

Reply:
Though, I guess, the reshape below will check for that

Reply:
Yeah, i think the reshape should be sufficient